File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    var obj = {};
    for(let i=0; i< yearning.length ; i++){
        obj[name[i]] = yearning[i];
    }
    console.log(obj);
    for(let i=0; i<photo.length; i++){
        let num = 0;
        for(let j=0; j<photo[i].length; j++){
            if(Object.keys(obj).includes( photo[i][j])){
                num += obj[photo[i][j]];
            }
            
        }
        answer.push(num);
    }
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    obj = {}
    for i in range(len(name)):
        obj[name[i]] = yearning[i]
    
    for i in photo:
        score = 0
        for j in i:
            if j in obj:
                score += obj[j]
        answer.append(score)
        
    return answer